src/ert/scheduler/job.py

```diff
@@ -78,9 +87,32 @@ async def __call__(
             await self.started.wait()
 
             await self._send(State.RUNNING)
-            while not self.returncode.done():
-                await asyncio.sleep(0.01)
-            returncode = await self.returncode
+
+            async def threadsafer_future_wait(future: asyncio.Future[_T]) -> _T:
+                while not future.done():
```

Comment:
I think this should be:
```python
try:
    while not future.done():
            await asyncio.sleep(0.01)
        returncode = await future
        return returncode
 except asyncio.CancelledError:
    timeout_event = CloudEvent(
                    {
                        "type": EVTYPE_REALIZATION_TIMEOUT,
                        "source": f"/ert/ensemble/{self._scheduler._ens_id}/real/{self.iens}",
                        "id": str(uuid.uuid1()),
                    }
                )
    assert self._scheduler._events is not None
    await self._scheduler._events.put(to_json(timeout_event))

```
`wait_for` automatically cancells the task that was waited_for.

Reply:
you want the except clause to happen inside the `threadsafer_future_wait` or do you want to avoid `threadsafer_future_wait`?

Reply:
It depends what we want to achieve on the cancellation due to the `max_runtime`. Based on the docs on `wait_for`, firstly the task gets cancelled, then the cancellation errors gets propagated and turns into the `TimeoutError`.